python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -8732,12 +8732,12 @@ def test_explode(data, labels, ignore_index, p_index, label_to_explode):
         (
             cudf.DataFrame({"a": [10, 0, 2], "b": [-10, 10, 1]}),
             True,
-            cudf.Series([1, 2, 0], dtype="int32"),
+            cupy.array([1, 2, 0], dtype="int32"),
```

Comment:
cc: @dantegd for visibility, we are making a breaking change as part of this PR, i.e., returning a cupy ndarray instead of a cudf.Series as the result of `argsort` calls.

Reply:
Are we? Skimming the implementation of `argsort` above, it looks  like we're still returning a `Series`.

Reply:
For `DataFrame` & `Index` this PR is returning `cupy.ndarray`, for `Series`, a `Series` is being returned.

Reply:
Yeah it's a cupy.array now, see [here](https://github.com/rapidsai/cudf/pull/9464/files#diff-45f163ddc7fb3fc5853e1fdd90a3d740a9dd0987c9ebcf00315ad7aff727576dR2969) where we're returning `_get_sorted_inds(...).values`. In my opinion it's more intuitive when returning integer indexes like this to also have them not be associated with any index and instead be suitable for iloc indexing, and `pd.DataFrame` doesn't support `argsort` (`pd.Series` does) so we have the flexibility to make such a choice. However we can revert this change if you two disagree or if this is break will cause problems for other RAPIDS libraries relying on the existing behavior.

Reply:
Probably also worth checking with cuGraph, CC @rlratzel.

Reply:
Thanks for the heads up @galipremsagar @vyasr , I just checked and this shouldn't affect the cuML codebase

Reply:
We're not using `argsort`. I also tested this branch against cugraph locally and the python tests passed, so I think this change is safe for us.

Reply:
If `pd.DataFrame` doesn't support `argsort`, can we:

1. Remove it for `cudf.DataFrame`?
2. and/or not test it?

Reply:
Unfortunately we can't remove it because we rely on it in our implementation of merging (and possibly in other places, but that's the obvious one that I'm aware of). I suppose that we could choose not to test it and rely on the `join(..., sort=True)` tests, but that would make our lives a little bit more difficult when it comes to identifying bugs.